### PR TITLE
Update README for Autolink plugin, in preparation for GA release

### PR DESCRIPTION
**Pull request by jasonblais**
@levb @jfrerich Tagging you as the reviewers of the Autolink plugin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
+    - If you are running Mattermost v5.11 or earlier, you must first go to the [releases page of this GitHub repository](https://github.com/mattermost/mattermost-plugin-autolink), download the latest release, and upload it to your Mattermost instance [following this documentation](https://docs.mattermost.com/administration/plugins.html#plugin-uploads).
+
+2. Modify your `config.json` file to include the types of regexp patterns you wish to match, under the `PluginSettings`. See below for an example of what this should look like.
 
 ## Usage
 
-Autolinks have 2 parts: a **Pattern** which is a regular expression search pattern utilizing the https://golang.org/pkg/regexp/ library, and a **Template** that gets expanded. You can create variables in the pattern with the syntax `(?P<name>...)` which will then be expanded by the corresponding template.
+Autolinks have 2 parts: a **Pattern** which is a regular expression search pattern utilizing the [Golang regexp library](https://golang.org/pkg/regexp/), and a **Template** that gets expanded. You can create variables in the pattern with the syntax `(?P<name>...)` which will then be expanded by the corresponding template.
```
Comment (hanzei) [inline README.md]: Please not that you can provided an optional `Scope` per link. See the first element of the example.

Comment (levb) [reply README.md]: A scope is a "Team Name[/Channel Name]", see https://github.com/mattermost/mattermost-plugin-autolink/blob/472c0ac69a15c9d7f6d75e48c85286972f442c4f/server/util.go#L9

Comment (jasonblais) [reply README.md]: Oh nice, didn't realize this was supported. Opened a separate PR to clarify this in docs https://github.com/mattermost/mattermost-plugin-autolink/issues/57

**Comment by jasonblais** [comment]
@levb @hanzei I opened #57 to document "Scope" more clearly. If this is good to merge, please merge because I don't have permissions to

**Pull request merged**